src/main/java/de/tum/in/www1/artemis/service/ExamService.java

```diff
@@ -550,9 +551,30 @@ public void validateForStudentExamGeneration(Exam exam) throws BadRequestAlertEx
         List<StudentDTO> notFoundStudentsDtos = new ArrayList<>();
         for (var studentDto : studentDtos) {
             var registrationNumber = studentDto.getRegistrationNumber();
+
+            // If no registration number is given try to fetch it via the login
+            if (!StringUtils.hasText(registrationNumber)) {
```

Comment:
I suggest that we do it in the following order:

1) Try to find the user based on the registration number (if available) in the database
2) Try to find the user based on the registration number (if available) in the LDAP service (if activated)
3) Try to find the user based on the login (if available) in the database

Only in case both values are available and the first two don't work, we would have two database calls, which is ok, as this is an edge case anyway.

A check if both values match is not needed in my opinion, as it would require some error handling that we would need to communicate to the instructor who initiated this request.
Of course this would be nice to have, but it could also be done in a follow up PR

Reply:
My idea here was that I save two database lookups with an empty number string in case the csv only specified a login. I'll change it to what you suggest though.